File: v1/src/main/java/com/api/v1/controllers/RentRegistrationController.java
package com.api.v1.controllers;

import com.api.v1.annotations.customer.SSN;
import com.api.v1.annotations.car.VIN;
import com.api.v1.dtos.rent.RentResponseDto;
import com.api.v1.services.rent.RentRegistrationService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("api/v1/rents")
public class RentRegistrationController {

    @Autowired
    private RentRegistrationService service;

    @PostMapping("{ssn}/{vin}/{paymentOrderNumber}/{days}")
    @ResponseStatus(value = HttpStatus.CREATED)
    public Mono<RentResponseDto> register(
            @PathVariable @SSN String ssn,
            @PathVariable @VIN String vin,
            @PathVariable String paymentOrderNumber,
            @PathVariable int days
    ) {
        return service.register(ssn, vin, paymentOrderNumber, days);
    }
}

File: kotlin/v2/src/main/kotlin/com/api/v2/rent/controllers/RentRegistrationController.kt
package com.api.v2.rent.controllers

import com.api.v2.rent.dtos.RentResponseDto
import com.api.v2.rent.service.RentRegistrationService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("api/v2/rents")
class RentRegistrationController {

    @Autowired
    private lateinit var service: RentRegistrationService

    @ResponseStatus(value = HttpStatus.CREATED)
    @PostMapping("{ssn}/{vin}/{paymentOrderNumber}/{days}")
    suspend fun register(
        @PathVariable ssn: String,
        @PathVariable vin: String,
        @PathVariable paymentOrderNumber: String,
        @PathVariable days: Long
    ): RentResponseDto {
        return service.register(ssn, vin, paymentOrderNumber, days)
    }

}